### PR TITLE
Ensure using of mem2reg option via library interface

### PR DESCRIPTION
**Pull request by imashkov**
-Implemented using of mem2reg option via TranslatorOpts

-Made setter for SPIRVGenKernelArgNameMD extension to remove this parameter
from TranslatorOpts constructor

Signed-off-by: Ilya Mashkov <ilya.mashkov@intel.com>

### CODE REVIEW

**Comment by imashkov** [comment]
@AlexeySotkin  @AlexeySachkov  Please, take a look

File: include/LLVMSPIRVLib.h [hunk 1]
```diff
@@ -149,8 +149,13 @@ std::unique_ptr<Module> convertSpirvToLLVM(LLVMContext &C,
 
 /// \brief Regularize LLVM module by removing entities not representable by
 /// SPIRV.
+
```
Comment (AlexeySachkov) [inline include/LLVMSPIRVLib.h:152]:
```suggestion
```

File: include/LLVMSPIRVLib.h [hunk 2]
```diff
@@ -149,8 +149,13 @@ std::unique_ptr<Module> convertSpirvToLLVM(LLVMContext &C,
...
 bool regularizeLlvmForSpirv(Module *M, std::string &ErrMsg);
 
+
```
Comment (AlexeySachkov) [inline include/LLVMSPIRVLib.h:155]:
```suggestion
```

File: include/LLVMSPIRVOpts.h [hunk 3]
```diff
@@ -128,6 +135,7 @@ class TranslatorOpts {
   ExtensionsStatusMap ExtStatusMap;
   // SPIR-V to LLVM translation options
   bool GenKernelArgNameMD = false;
+  bool SPIRVMemToReg = false;
```
Comment (AlexeySachkov) [inline include/LLVMSPIRVOpts.h:138]: Lines 137-142 describes SPIR-V to LLVM translation options

I suggest to move this new option somewhere out of this block and left a comment that this option affects LLVM IR regularization phase

File: test/mem2reg.cl [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// RUN: %clang_cc1 -O0 -S -triple spir-unknown-unknown -cl-std=CL2.0 -x cl -disable-O0-optnone %s -emit-llvm-bc -o %t.bc
+// RUN: llvm-spirv -s %t.bc
+// RUN: llvm-dis < %t.bc | FileCheck %s --check-prefixes=CHECK,CHECK-W
+// RUN: llvm-spirv -s -spirv-mem2reg %t.bc -o %t.opt.bc
+// RUN: llvm-dis < %t.opt.bc | FileCheck %s --check-prefixes=CHECK,CHECK-WO
```
Comment (AlexeySachkov) [inline test/mem2reg.cl:5]: It is probably better swap names `CHECK-W` and `CHECK-WO`, i.e. `WO` stands for "without mem2reg", `W` stands for "with mem2reg"

**Comment by imashkov** [comment]
@AlexeySachkov Thanks for review, I applied your suggestions

File: lib/SPIRV/SPIRVWriter.cpp [hunk 5]
```diff
@@ -2622,12 +2620,21 @@ bool llvm::writeSpirv(Module *M, const SPIRV::TranslatorOpts &Opts,
 }
 
 bool llvm::regularizeLlvmForSpirv(Module *M, std::string &ErrMsg) {
+  SPIRV::TranslatorOpts DefaultOpts;
+  // To preserve old behavior of the translator, let's enable all extensions
+  // by default in this API
+  DefaultOpts.enableAllExtensions();
+  llvm::regularizeLlvmForSpirv(M, ErrMsg, DefaultOpts);
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVWriter.cpp:2627]:
```suggestion
  return llvm::regularizeLlvmForSpirv(M, ErrMsg, DefaultOpts);
```

Comment (imashkov) [reply lib/SPIRV/SPIRVWriter.cpp]: Thanks, I done this.

**Pull request merged**